### ISSUE TITLE
Peerstore not being closed properly with the host

### ISSUE THREAD

**Issue by MichaelMure** [libp2p/go-libp2p#1031]
Found with https://github.com/uber-go/goleak.

To reproduce, start a libp2p node, possibly add gossip-sub, cancel the context. The following goroutine should still be running:

It looks like the `Close()` function of the Peerstore is not called properly but I can't pinpoint where exactly.

```
Goroutine 3 in state select, 1 minutes, with github.com/libp2p/go-libp2p-peerstore/pstoremem.(*memoryAddrBook).background on top of the stack:
goroutine 3 [select, 1 minutes]:
github.com/libp2p/go-libp2p-peerstore/pstoremem.(*memoryAddrBook).background(0xc000192000)
	/home/michael/go/pkg/mod/github.com/libp2p/go-libp2p-peerstore@v0.2.6/pstoremem/addr_book.go:93 +0x126
created by github.com/libp2p/go-libp2p-peerstore/pstoremem.NewAddrBook
	/home/michael/go/pkg/mod/github.com/libp2p/go-libp2p-peerstore@v0.2.6/pstoremem/addr_book.go:83 +0x18f
```

**Comment by MichaelMure** [comment]
More details at https://github.com/libp2p/go-libp2p/issues/1030#issuecomment-748069158

**Comment by jacobheun** [comment]
Pending fix at https://github.com/libp2p/go-libp2p/pull/1037

**Comment by marten-seemann** [comment]
Was this fixed by #1037? Can we close this issue?

**Comment by aarshkshah1992** [comment]
@marten-seemann Yes, this issue can be closed.
